src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,45 @@
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private static final String DEFAULT_DELIMITER = "[,:]";
+
+    public static int splitAndSum(String text) {
+        if (text == null) {
+            return 0;
+        }
+
+        if (text.isEmpty()) {
+            return 0;
+        }
+
+        if (text.length() == 1) {
+            return parseInt(text);
+        }
+
+        return sum(split(text));
+    }
+
+    private static int parseInt(String text) {
+        try {
+            return Integer.parseUnsignedInt(text);
+        }catch (Exception e) {
+            throw new RuntimeException();
+        }
+    }
+
+    private static String[] split(String text) {
+        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(text);
```

Comment:
효진님에게 드렸던 피드백 내용과 같아 복붙시전하겠습니다. 😅

---

계산기를 호출할 때 마다 정규식을 매번 컴파일하겠네요.
정규식 비용은 비싸기 때문에 고정된 값이라면 패턴 자체를 상수화시키는 것이 좋습니다.

<img width="585" alt="image" src="https://github.com/g-oop/goop-java-racingcar/assets/36617783/aac31e95-968e-4466-9f25-a1f7846f932a">

```suggestion
        Matcher matcher = PATTERN.matcher(text);
```